### DISCUSSION TITLE
Duda sobre la gestión

### DISCUSSION THREAD

**Discussion by BrahianCarrera** [rramosp/ai4eng.v1#310] [04 - GESTIÓN DE DATOS]
Hola, tengo una duda con respecto al dataframe del proyecto final, he visto que es recomendable convertir algunas variables categóricas en one-hot para no sesgar los datos. Mi preguntas son: 

Hay alguna convención de cuando conviene y cuando no convertir las variables categóricas o para los SVM obligatoriamente toca?

Además, Algunas columnas del DF tienen muchas variables categóricas, es rentable convertir la columna del programa universitario por ejemplo que cuenta con muchas categorías ? Me preocupa solo por que quedaría un dataframe bastante amplio 

Muchas gracias por su tiempo

**Comment by Felipe-RA** [comment]
Hola @BrahianCarrera 

Gracias por compartir tus dudas por acá!

Lo primero, todas estas respuestas que te voy a dar van con el **disclaimer** de que la mejor forma de saber lo que funciona bien o no es probar!!

Con eso dicho, algunas guías generales:

1. **Sobre el One-Hot Encoding: ** Generalmente utilizas el One-Hot encoding cuando las categorías no tienen un órden inherente o jerarquía que debas preservar. Esto es especialmente útil para el caso de un **SVM** puesto que asume las entradas de datos como numéricas. Si necesitases convertir datos ordinales, está la técnica de [ordinal encoding](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OrdinalEncoder.html)
2. **¿Qué hacer cuando tienes variables con muchas categorías?: ** En estos casos, (y de nuevo, superditado a que hagas las pruebas pertinentes) lo que tienes es un caso en el que vas a balancear entre el aumento de la dimensionalidad de tus datos (y el costo computacional añadido que esto implica) vs el posible beneficio que puedas obtener (e incluso, al llegar a altos niveles de dimensionalidad te puedes encontrar el caso de la llamada "maldición de la dimensionalidad", en la que tu modelo pierde la capacidad de generalizar por la dispersión artificialmente inflada de los datos)


Ahora, ¿qué hacer en el caso del punto 2? Técnicas que busquen reducir la dimensionalidad

1.  Limpiar los datos para que categorías que son en realidad las mismas no sean tomadas como distintas categorías
2.  Reducir el número de categorías,  un criterio simple que puedes usar es meter en una categoría genérica ("OTROS") las categorías que no tengan muchas repeticiones en tu dataset (el umbral adecuado es, nuevamente, algo a lo que te puedes acercar experimentando)


Hasta aquí son los consejos generales, luego puedes probar "ideas locas" y ver si ayudan con el proceso (nuevamente, experimentando). Te comparto una de forma suelta, que te puede ayudar o no, ¿y si buscas lo que es un **lasso** y pruebas a ayudarte de este para reducir las categorías? Esto te puede funcionar o no, lo importante nuevamente es que cuantifiques el resultado y mires que tal te fue!

¡Estoy pendiente cualqueir otra cosa, hasta pronto!
